src/v/raft/replicate_entries_stm.cc

```diff
@@ -361,12 +368,14 @@ ss::future<> replicate_entries_stm::wait() { return _req_bg.close(); }
 replicate_entries_stm::replicate_entries_stm(
```

Comment:
Would that be possible to  split the `replicate_stm::apply` method into parts ? we may add a method like: `dispatch()`, `wait_for_leader()`, `wait_for_majority()`. This way we wouldn't have to deal with the promise passed by the caller ?